Developer Docs: Include Guide (#874)
ECP-WarpX/WarpX: Docs/source/developers/repo_organization.rst

Build system
------------

WarpX uses the AMReX build system (GNUMake).

> Each sub-folder contains a file ``Make.package`` with the names of source files to add for the build.

Reviewer: Very very minor (possible - feel free to re-write or ignore) suggestion:
```suggestion
Each sub-folder contains a file ``Make.package`` with the names of source files that are added for the build.
```

> Each sub-folder contains a file ``Make.package`` with the names of source files that are added to the build.

Reviewer 2: Great PR! I would just suggest to be more explicit here about the fact that header files should not appear in the `Make.package` files. I would add something like ``Make.package should contain source files only (*.cpp). Header files (*.H) should not be listed in Make.package``

C++ Includes
------------

All WarpX header files need to be specified relative to the ``Source/`` directory.

- e.g. ``#include "Utils/WarpXConst.H"``
- files in the same directory as the including header-file can be included with ``#include "FileName.H"``

The include order and proper quotation marks are:

> - WarpX header files ``#include "..."`` then
> - AMReX header files ``#include <...>`` then
> - other third party includes ``#include <...>`` then
> - standard library includes, e.g. ``#include <vector>``

Reviewer 3: Just a suggestion, as you are presenting an include order.
```suggestion
1. WarpX header files ``#include "..."`` then
2. AMReX header files ``#include <...>`` then
3. other third party includes ``#include <...>`` then
4. standard library includes, e.g. ``#include <vector>``
```
